pinot-core/src/main/java/org/apache/pinot/core/realtime/stream/MessageBatch.java

```diff
@@ -18,11 +18,17 @@
  */
 package org.apache.pinot.core.realtime.stream;
 
+import org.apache.pinot.annotations.InterfaceAudience;
+import org.apache.pinot.annotations.InterfaceStability;
+
+
 /**
  * Interface wrapping stream consumer. Throws IndexOutOfBoundsException when trying to access a message at an
  * invalid index.
  * @param <T>
  */
+@InterfaceAudience.Public
+@InterfaceStability.Stable
```

Comment:
I would call this evolving at this time, until we get another stream in place.

Reply:
Based on this definition of stable:

"Can evolve while retaining compatibility for minor release boundaries.;
   * can break compatibility only at major release (ie. at m.0)."

and the fact that we expect external users to extend this interface, I think we can call it stable. Evolving annotation would be used for anything we are still figuring out - for ex, a first draft of an interface.

Reply:
agreed. This is a note to ourselves that since there may be users who have implemented this interface but are not able to contribute the code here for whatever reason, and we don't want to break their code as  best as we can